contracts/contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -449,15 +451,26 @@ namespace eosiosystem {
    void system_contract::undelegatebw( const name& from, const name& receiver,
                                        const asset& unstake_net_quantity, const asset& unstake_cpu_quantity )
    {
-      asset zero_asset( 0, core_symbol() );
-      check( unstake_cpu_quantity >= zero_asset, "must unstake a positive amount" );
-      check( unstake_net_quantity >= zero_asset, "must unstake a positive amount" );
-      check( unstake_cpu_quantity.amount + unstake_net_quantity.amount > 0, "must unstake a positive amount" );
-      check( _gstate.total_activated_stake >= min_activated_stake,
-             "cannot undelegate bandwidth until the chain is activated (at least 15% of all tokens participate in voting)" );
-
-      changebw( from, receiver, -unstake_net_quantity, -unstake_cpu_quantity, false);
-   } // undelegatebw
+
+        if(is_block_producer(from))
```

Comment:
@AndriiAndrusyshyn It's better to place this validation after EOS checks so we will have less diff